### PR TITLE
Implements JSON Web Signature

### PR DESCRIPTION
**Pull request by antiphoton**
* Implements JSON Web Signature, which is [needed](https://datatracker.ietf.org/doc/html/rfc8555#:~:text=payload%20in%20a%20JSON%20Web%20Signature) by ACME request body.
* Add a serde helper to serialize the field of `EcPublicKey` into base64 format.

### CODE REVIEW

File: sxg_rs/src/acme/jws.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+/// The protected headers which is used authentication ACME request
+/// [(ACME spec)](https://datatracker.ietf.org/doc/html/rfc8555#:~:text=The%20JWS%20Protected%20Header%20MUST%20include%20the%20following%20fields).
+#[derive(Serialize)]
+struct AcmeProtectedHeader<'a> {
+    // https://datatracker.ietf.org/doc/html/rfc7518#section-3.1
+    alg: &'static str,
+    nonce: String,
+    url: &'a str,
+    jwk: Option<&'a EcPublicKey>,
+    kid: Option<&'a str>,
```
Comment (banaag) [inline sxg_rs/src/acme/jws.rs:53]: Does Cloudflare support EAB (External account binding) (see: https://github.com/go-acme/lego/blob/master/acme/api/account.go#L31).

File: sxg_rs/src/acme/jws.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module implements
+//! [JSON Web Signature (JWS)](https://datatracker.ietf.org/doc/html/rfc7515).
+//! [ACME spec](https://datatracker.ietf.org/doc/html/rfc8555#:~:text=payload%20in%20a%20JSON%20Web%20Signature)
+//! requires the request body to be encapsulated in JWS format for authenication.
```
Comment (banaag) [inline sxg_rs/src/acme/jws.rs:18]: authenication -> authentication

File: sxg_rs/src/acme/jws.rs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+use crate::crypto::EcPublicKey;
+use crate::signature::Signer;
+use anyhow::{Error, Result};
+use serde::Serialize;
+
+pub async fn create_acme_request_body<S: Signer, P: Serialize>(
```
Comment (banaag) [inline sxg_rs/src/acme/jws.rs:25]: Where is this called?

File: sxg_rs/src/acme/jws.rs (hunk 2 above)
Comment (antiphoton) [reply sxg_rs/src/acme/jws.rs]: Fixed

File: sxg_rs/src/acme/jws.rs (hunk 3 above)
Comment (antiphoton) [reply sxg_rs/src/acme/jws.rs:25]: This PR only focuses on JWS. A following up PR of ACME will use this function.

**Review by banaag — APPROVED** [review]

**Pull request merged**